File: app/src/main/java/com/nearsoft/nearbooks/ws/responses/AvailabilityResponse.java
package com.nearsoft.nearbooks.ws.responses;

import com.google.gson.annotations.SerializedName;
import com.nearsoft.nearbooks.models.sqlite.Book;
import com.nearsoft.nearbooks.models.sqlite.Borrow;

import java.util.Date;

/**
 * Availability response.
 * Created by epool on 1/12/16.
 */
public class AvailabilityResponse {

    @SerializedName("codeQRID")
    private String codeQrId;

    @SerializedName("bookID")
    private String bookId;

    @SerializedName("copyNumber")
    private int copyNumber;

    @SerializedName("isAvailable")
    private boolean isAvailable;

    @SerializedName("nextStatus")
    private int nextStatus;

    @SerializedName("initialDateAvailability")
    private Date initialDateAvailability;

    @SerializedName("finalDateAvailability")
    private Date finalDateAvailability;

    @SerializedName("message")
    private String message;

    @SerializedName("book")
    private Book book;

    public String getCodeQrId() {
        return codeQrId;
    }

    public void setCodeQrId(String codeQrId) {
        this.codeQrId = codeQrId;
    }

    public String getBookId() {
        return bookId;
    }

    public void setBookId(String bookId) {
        this.bookId = bookId;
    }

    public int getCopyNumber() {
        return copyNumber;
    }

    public void setCopyNumber(int copyNumber) {
        this.copyNumber = copyNumber;
    }

    public boolean isAvailable() {
        return isAvailable;
    }

    public void setAvailable(boolean available) {
        isAvailable = available;
    }

    @Borrow.Status
    public int getNextStatus() {
        return nextStatus;
    }

    public void setNextStatus(@Borrow.Status int nextStatus) {
        this.nextStatus = nextStatus;
    }

    public Date getInitialDateAvailability() {
        return initialDateAvailability;
    }

    public void setInitialDateAvailability(Date initialDateAvailability) {
        this.initialDateAvailability = initialDateAvailability;
    }

    public Date getFinalDateAvailability() {
        return finalDateAvailability;
    }

    public void setFinalDateAvailability(Date finalDateAvailability) {
        this.finalDateAvailability = finalDateAvailability;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public Book getBook() {
        return book;
    }

    public void setBook(Book book) {
        this.book = book;
    }

}

File: app/src/main/java/com/nearsoft/nearbooks/ws/responses/AvailabilityResponse.kt
package com.nearsoft.nearbooks.ws.responses

import com.google.gson.annotations.SerializedName
import com.nearsoft.nearbooks.models.realm.Book
import com.nearsoft.nearbooks.models.realm.Borrow
import java.util.*

/**
 * Availability response.
 * Created by epool on 1/12/16.
 */
data class AvailabilityResponse(
        @SerializedName("codeQRID") val codeQrId: String? = null,
        @SerializedName("bookID") val bookId: String? = null,
        @SerializedName("copyNumber") val copyNumber: Int,
        @SerializedName("isAvailable") val isAvailable: Boolean,
        @SerializedName("nextStatus") val nextStatus: Int,
        @SerializedName("initialDateAvailability") val initialDateAvailability: Date? = null,
        @SerializedName("finalDateAvailability") val finalDateAvailability: Date? = null,
        @SerializedName("message") val message: String? = null,
        @SerializedName("activeBorrrow") val activeBorrow: Borrow? = null,
        @SerializedName("book") val book: Book? = null
)
